packages/rainbowkit/src/wallets/Wallet.ts

```diff
@@ -70,23 +70,34 @@ export type Wallet<C extends Connector = Connector> = {
     linux?: string;
     desktop?: string;
   };
-  hidden?: (args: {
-    wallets: {
-      id: string;
-      connector: Connector;
-      installed?: boolean;
-      name: string;
-    }[];
-  }) => boolean;
-  createConnector: () => RainbowKitConnector<C>;
-};
+  hidden?: () => boolean;
+  createConnector: (options: WalletOptionsParams) => CreateConnectorFn;
+} & RainbowKitConnector;
 
 export type WalletList = { groupName: string; wallets: Wallet[] }[];
 
-export type WalletInstance = Omit<Wallet, 'createConnector' | 'hidden'> &
-  ReturnType<Wallet['createConnector']> & {
-    index: number;
-    groupIndex: number;
-    groupName: string;
-    walletConnectModalConnector?: Connector;
-  };
+export type RainbowKitDetails = Omit<Wallet, 'createConnector' | 'hidden'> & {
+  index: number;
+  groupIndex: number;
+  groupName: string;
+  isWalletConnectModalConnector?: boolean;
+  isRainbowKitConnector: boolean;
+  walletConnectModalConnector?: Connector;
+  showQrModal?: boolean;
+};
+
+export type WalletOptionsParams = { rkDetails: RainbowKitDetails };
+
+export type CreateConnector = (walletOptions: {
+  rkDetails: RainbowKitDetails;
+}) => CreateConnectorFn;
+
+// This is the default connector you get at first from wagmi
+// "Connector" + rainbowkit details we inject into the connector
+export type WagmiConnectorInstance = Connector & {
+  rkDetails: RainbowKitDetails;
+};
+
```

Comment:
I recently added `rkDetails` to the connectors. It's not the best, but it was really needed. Skipping `rkDetails` could lead to bugs, like wallet connecting and disconnecting after 5 seconds randomly. This happens because we're overriding the object like this:

```
createConnector((config) => ({
    ...injected(injectedConfig)(config),
    ...walletOptions // If `rkDetails` not passed here it will introduce bugs
}));
```

By implementing `rkDetails`, we avoid these bugs and can put our own icons and download links in the `useWalletConnectors`  hook. This is particularly important since wagmi uses emitters and tanstack. I've also taken a look and without `rkDetails` they will include our values in their functions which is not good.